### PR TITLE
auth gubernator

### PR DESCRIPTION
**Pull request by krzyzacy**
for https://github.com/kubernetes/test-infra/issues/6986

and I finally get it right!
deployed internally already

/area gubernator
/assign @rmmh @BenTheElder

### CODE REVIEW

**Comment by krzyzacy** [comment]
hummm diff is 
```
......ERROR:root:unable to fetch 'https://www.googleapis.com/storage/v1/b/kubernetes-jenkins/o?delimiter=/&prefix=pr-logs/pull/12345/': status code 401
```

assume I'll need to mock out something else? will take another look :-\

File: gubernator/gcs_async.py [hunk 1]
```diff
@@ -26,7 +27,17 @@
 @ndb.tasklet
 def get(url):
     context = ndb.get_context()
-    headers = {'accept-encoding': 'gzip, *', 'x-goog-api-version': '2'}
+
+    headers = {
```
Comment (rmmh) [inline gubernator/gcs_async.py]: This code should only execute if the URL is a cloud endpoint. You can use urlparse.urlparse to check the domain against a whitelist.

Comment (krzyzacy) [reply gubernator/gcs_async.py]: hummm, but why the test failed?

File: gubernator/gcs_async.py [hunk 2]
```diff
@@ -26,7 +28,19 @@
...
+        'accept-encoding': 'gzip, *',
+        'x-goog-api-version': '2',
+        }
+
+    result = urlparse.urlparse(url)
+    if 'googleapis.com' in result.netloc:
```
Comment (rmmh) [inline gubernator/gcs_async.py]: if result.netloc.endswith('.googleapis.com')

File: gubernator/gcs_async.py [hunk 3]
```diff
@@ -26,7 +28,19 @@
...
     context = ndb.get_context()
-    headers = {'accept-encoding': 'gzip, *', 'x-goog-api-version': '2'}
+
+    headers = {
+        'accept-encoding': 'gzip, *',
+        'x-goog-api-version': '2',
+        }
+
+    url_result = urlparse.urlparse(url)
+    if url_result.netloc.endswith('googleapis.com'):
```
Comment (rmmh) [inline gubernator/gcs_async.py]: Needs to be ".googleapis.com". This might be paranoia...

**Comment by rmmh** [comment]
It looks like the Authorization header makes the mocked GCS instance fail to respond :\

**Comment by krzyzacy** [comment]
:-\ I cannot push from my laptop anymore (?)

**Comment by krzyzacy** [comment]
wait, why it tries to pull from the actual storage endpoint, rather than the `http://testbed.example.com` one?

File: gubernator/view_pr_test.py [hunk 4]
```diff
@@ -194,6 +194,8 @@ def test_me(self):
 
     def test_pr_links_user(self):
         "Individual PR pages grab digest information"
+        gcs_async_test.install_handler(self.testbed.get_stub('urlfetch'),
+            {'12345/': []})
```
Comment (krzyzacy) [inline gubernator/view_pr_test.py]: @rmmh ok this worked - I think it was trying to pull from the actual bucket

File: gubernator/gcs_async.py [hunk 5]
```diff
@@ -26,7 +28,19 @@
...
     context = ndb.get_context()
-    headers = {'accept-encoding': 'gzip, *', 'x-goog-api-version': '2'}
+
+    headers = {
+        'accept-encoding': 'gzip, *',
+        'x-goog-api-version': '2',
+        }
+
+    url_result = urlparse.urlparse(url)
+    if url_result.netloc.endswith('.googleapis.com'):
```
Comment (rmmh) [inline gubernator/gcs_async.py]: I don't know how to make the stub happy with this. You could guard it all with `if 'testbed' not in os.environ.get('SERVER_SOFTWARE', 'testbed'):`.

Gross :(

Comment (krzyzacy) [reply gubernator/gcs_async.py]: see my previous comment, I think `install_handler` need to be explicitly called for `listdirs`?

**Pull request merged**